analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -147,31 +130,32 @@ private static bool HasMutableUsagesInMethod(VariableDeclaratorSyntax variable,
                 parentSyntax = parentSyntax.Parent;
             }
 
+            ISymbol variableSymbol = null;
+
             return parentSyntax
                 .DescendantNodes()
                 .OfType<IdentifierNameSyntax>()
                 .Where(MatchesIdentifier)
                 .Any(x => IsMutatingUse(semanticModel, x));
 
             static bool IsMethodLike(SyntaxNode arg) =>
-                arg is BaseMethodDeclarationSyntax
-                || arg is IndexerDeclarationSyntax
-                || arg is AccessorDeclarationSyntax
-                || arg is LambdaExpressionSyntax
-                || arg is GlobalStatementSyntax;
-
-            bool MatchesIdentifier(IdentifierNameSyntax id) =>
-                Equals(variableSymbol, semanticModel.GetSymbolInfo(id).Symbol);
+                arg is BaseMethodDeclarationSyntax or IndexerDeclarationSyntax or AccessorDeclarationSyntax or LambdaExpressionSyntax or GlobalStatementSyntax;
+
+            bool MatchesIdentifier(IdentifierNameSyntax id)
+            {
+                variableSymbol ??= semanticModel.GetDeclaredSymbol(variable);
```

Comment:
The lazy loading here is questionable. The likelihood of finding an identifier within a method is close to 100%:
```cs
 return parentSyntax
                .DescendantNodes()
                .OfType<IdentifierNameSyntax>()
                .Where(MatchesIdentifier)
                .Any(x => IsMutatingUse(semanticModel, x));
```